src/Switch/index.jsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2017-2019 dunnhumby Germany GmbH.
+ * All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the LICENSE file
+ * in the root directory of this source tree.
+ *
+ */
+
+import React        from 'react';
+import PropTypes    from 'prop-types';
+
+import {
+    attachEvents,
+    useId,
+    useTheme,
+} from '../utils';
+
+
+const componentName = 'Switch';
+
+const Switch = props =>
+{
+    const {
+        isChecked,
+        isDefaultChecked,
+        isDisabled,
+        label,
+        onMouseOut,
+        onMouseOver,
+    } = props;
+
+    const cssMap = useTheme( componentName, props );
+    const id = useId( componentName, props );
+
+    return (
+        <div
+            className    = { cssMap.main }
+            onMouseEnter = { onMouseOver }
+            onMouseLeave = { onMouseOut }>
+            <input
+                { ...attachEvents( props ) }
```

Comment:
The idea of the `attachEvents` helper is that it should go on the outer element of the component:
```
const {
    isChecked,
    isDefaultChecked,
    isDisabled,
    label,
} = props;
...
return (
    <div { ...attachEvents( props ) } className = { cssMap.main }>
        <input
             checked        = { isChecked }
             className      = { cssMap.input }
             disabled       = { isDisabled }
             id             = { id }
             defaultChecked = { isDefaultChecked }
             type           = "checkbox" />
        ...
    </div>
);
```
We need to improve the whole this whole confusing `attachEvents` business but that’s another issue. Sorry about that.